lib/howl/ui/editor.moon

```diff
@@ -367,16 +367,34 @@ class Editor extends PropertyObject
       return
 
     conf = @config_at_cursor
-    if conf.tab_indents and @current_context.prefix.is_blank
-      cur_line = @current_line
-      next_indent = cur_line.indentation + conf.indent
-      next_indent -= (next_indent % conf.indent)
-      cur_line.indentation = next_indent
-      @cursor.column = next_indent + 1
-    else if conf.use_tabs
+
+    if conf.use_tabs
```

Comment:
When `tab_indents` is true and where in whitespace we should not insert any tab, so this would break the current and desired behaviour.

Reply:
This comment has not been addressed, and the behaviour with this PR remains broken. A recap of what this function should do: 

If we're in leading whitespace and `tab_indents` is true, we should indent and nothing else. Whether tabs are used in that case or not is already handled when setting a buffer line's indentation.

If the above is not true, then we should insert a tab, or a suitable number spaces. The current implementation of this is very straightforward: one tab, or <space> * `indent`

In an attempt to better align with tab widths it would then be desirable to make that last part smarter, by optionally inserting fewer spaces to make it align with a virtual tab stop. That seems like it should only require modifications for the last `else` clause in the current implementation, unless one also wants to skip tabs even when `use_tabs` is true in order to align correctly.

As it stands, I think the current implementation is correct and rather clear given the above requirements, but could be smarter for that last case. The PR now however makes it incorrect, and more unclear as to what the behaviour should be. I suggest just looking at modifying the last `else` clause and see if that is enough.

Reply:
Alright, I'll revisit after Jul and hopefully get it almost completely right the next commit XD